src/components/internal/FormFields/FileInput/components/Dropzone.tsx

```diff
@@ -115,7 +120,13 @@ export const Dropzone = fixedForwardRef<DropzoneProps, HTMLInputElement>((props,
                     if (!allowedFileTypes.includes(file.type)) {
                         throw validationErrors.DISALLOWED_FILE_TYPE;
                     }
-                    if (file.size > maxFileSize) {
+
+                    // Determine the current max file size for the file type
+                    const currentMaxFileSize = typeof maxFileSize === 'function' ? maxFileSize(file.type) || DEFAULT_MAX_FILE_SIZE : maxFileSize;
+
+                    if (file.size > currentMaxFileSize) {
+                        const type = file.type.replace('image/', '').replace('application/', '');
+                        setLargeFileErrorContext({ type: type, limit: getHumanReadableFileSize(currentMaxFileSize) });
```

Comment:
This here seems opinionated, to match the current use case. I'd suggest sending the complete file type.
```suggestion
                        setLargeFileErrorContext({ type: file.type, limit: getHumanReadableFileSize(currentMaxFileSize) });
```